### PR TITLE
[NO JIRA] Log if the metrics server stared successfully

### PR DESCRIPTION
**Pull request by astrozzc**
## Link(s) to Jira
-

## Description of Intent of Change(s)
We have to ensure the worker does not start if the metrics server is not up.
And we have to log if the metrics server started successfully.

## Local Testing
How can the feature be exercised?
How can the bug be exploited and fix confirmed?
Is any special local setup required?

## Checklist
- [ ] if API spec changes are required, is the spec updated?
- [ ] are there any pre/post merge actions required? if so, document here.
- [ ] are theses changes covered by unit tests?
- [ ] if warranted, are documentation changes accounted for?
- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?
- [ ] is there known, direct impact to dependent teams/components?
  - [ ] if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: rbac/rbac/celery.py [hunk 1]
```diff
@@ -73,4 +79,14 @@ def start_metrics_server(sender=None, **kwargs):
     """Start the metrics server."""
     registry = CollectorRegistry()
     multiprocess.MultiProcessCollector(registry)
-    start_http_server(LoadedConfig.metricsPort, addr="0.0.0.0", registry=registry)
+    metrics_port = LoadedConfig.metricsPort or 9000
+
+    try:
+        start_http_server(metrics_port, addr="0.0.0.0", registry=registry)
+        logger.info(f"Server started and exposing to port {metrics_port}.")
+        raise Exception("test exception")
```
Comment (alechenninger) [inline rbac/rbac/celery.py:87]:
```suggestion
```

File: rbac/rbac/celery.py [hunk 2]
```diff
@@ -73,4 +79,14 @@ def start_metrics_server(sender=None, **kwargs):
...
+    except Exception as e:
+        sentry_sdk.capture_exception(e)
+        logger.error(f"Failed to start metrics server: {e}")
+        # Exit the entire process, we don't wat to spin up the worker without metrics
+        sys.exit()
```
Comment (alechenninger) [inline rbac/rbac/celery.py:92]: Maybe non-zero code to indicate error?

```suggestion
        sys.exit(1)
```

File: rbac/rbac/celery.py (hunk 1 above)
Comment (astrozzc) [reply rbac/rbac/celery.py]: oops, good catch! This for for testing locally 😅

File: rbac/rbac/celery.py (hunk 2 above)
Comment (astrozzc) [reply rbac/rbac/celery.py]: Updated

**Review by alechenninger — APPROVED** [review]

**Pull request merged**